generics/middleware/authenticator.js

```diff
@@ -417,8 +509,67 @@ module.exports = async function (req, res, next) {
       }
     }
   }
+
+
+    if (performInternalAccessTokenCheck) {
+      // Validate the presence of required headers
+      let adminOrgId = req.get('orgId');
+      let adminTenantId = req.get('tenantId');
+      let roles = decodedToken.data.roles;
+      let isAdmin = req.get('admin-auth-token') === process.env.ADMIN_AUTH_TOKEN;
+      let isTenantAdmin = checkForRole(roles, 'tenant_admin');
+      let isOrgAdmin = checkForRole(roles, 'org_admin');
+
+      if (isAdmin) {
```

Comment:
@borkarsaish65 For me the logic is simple. if you get orgId and teneant id passed via header you do all the role check, org validation, if not you always proceed with data from the token  &  internal access check you have to do. tell me if it not work. @Prajwal17Tunerlabs what do you think. please recheck the current logic work with all cases. consider a tenant or org admin trying to access portal to verify the solutions coming up correctly in portal

Reply:
I checked all the usecase it is working. 
The approach u said where we will only check if orgId and tenantId is passed , what happens if they dont pass and try to access api like program create , it will give an error in the code level. 
The current logic ensures that does not happen till the user login belongs to super admin or tenant_admin or org_admin.